science/apbs/Portfile

```diff
@@ -20,64 +17,86 @@ long_description        APBS is a software package for the numerical \
                         electrostatic interactions between molecular \
                         solutes over a wide range of length scales.
 
-github.setup            Electrostatics apbs-pdb2pqr d4e78c62e6a07f92c26924318d83a159fa6af063
-version                 1.4.1
-homepage                http://www.poissonboltzmann.org
-worksrcdir              ${worksrcdir}/apbs
-revision                6
-
-# stealth update; remove for next version
-dist_subdir             ${name}/${version}_1
-checksums               rmd160  09ffa6a90c190eb4e584934519c8cbd6229fc7b8 \
-                        sha256  3c3e6914a1a69c27d54d9d9ea7ffd1072e3bc82209971289f0a8a7964fcdf82b \
-                        size    37120958
-
-depends_lib             port:maloc \
-                        port:readline
-
-# dependencies are not managed correctly within this code, so errors such as from -lapbs_geoflow might occur
-# how to solve: first 'make apbs_geoflow', then 'make all'
-use_parallel_build no
-
-configure.optflags      -O3 -ffast-math
-configure.args-append   -DENABLE_OPENMP:BOOL=OFF \
-                        -DENABLE_BEM=OFF \
-                        -DCMAKE_C_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cflags}" \
-                        -DCMAKE_CXX_COMPILER_ARG1:STRING="-I${worksrcpath}/include ${configure.cxxflags} -stdlib=libc++"
-
-# otherwise obsolete subport installation will fail on patch phase and not get to pre-configure error about replacement
-if {${subport} eq ${name}} {
-    patchfiles              patch-CMakeLists.txt.diff patch-Eigen-include-path.diff patch-contrib-tabipb-CMakeLists.txt.diff patch-pbsolvercz.cpp.diff
-    if {[mpi_variant_isset]} {
-        patchfiles-append   patch-tests-apbs_tester.py.diff
+version                 3.4.1
+homepage                https://www.poissonboltzmann.org
+revision                0
+worksrcdir              ${workpath}/apbs-${version}
+
+extract.mkdir           no
+master_sites            https://github.com/Electrostatics/FETK/archive/refs/tags/ \
+                        https://github.com/Electrostatics/apbs/archive/refs/tags/
+distfiles               v${version}.tar.gz 1.9.3.tar.gz 
```

Comment:
Using distfiles for two different projects and not including the project name in either of them is a recipe for disaster. Please include the project name in both distfile names, as the github portgroup would do for the main distfile by default. This is easy to do at GitHub since automatically-generated tarballs can be requested by any filename, as long as the tag name is correct. The Aseprite port is an example of how to download distfiles from two different GitHub repositories.

Reply:
Switched to downloading named tarballs by GitHub commit for the specific version releases.

Reply:
Why use commit hashes? Isn't it more straightforward to use tag names like we do in other ports that download from GitHub?

Reply:
Updated the Portfile to use tagged releases.